src/services/suggestionDiagnostics.ts

```diff
@@ -171,16 +171,30 @@ namespace ts {
     }
 
     // should be kept up to date with getTransformationBody in convertToAsyncFunction.ts
-    function isFixablePromiseArgument(arg: Expression): boolean {
+    function isFixablePromiseArgument(arg: Expression, checker: TypeChecker): boolean {
         switch (arg.kind) {
             case SyntaxKind.FunctionDeclaration:
             case SyntaxKind.FunctionExpression:
             case SyntaxKind.ArrowFunction:
                 visitedNestedConvertibleFunctions.set(getKeyFromNode(arg as FunctionLikeDeclaration), true);
                 // falls through
             case SyntaxKind.NullKeyword:
-            case SyntaxKind.Identifier: // identifier includes undefined
                 return true;
+            case SyntaxKind.Identifier:
+            case SyntaxKind.PropertyAccessExpression: {
+                const symbol = checker.getSymbolAtLocation(arg);
+                if (!symbol) {
+                    return false;
+                }
+                if (checker.isUndefinedSymbol(symbol)) {

```

Comment:
how could a property access have the symbol of `undefined`? If it can't, why are some identifiers now not fixable?

Reply:
`property access` cannot be, however, Identifier can be., `case SyntaxKind.Identifier:` was used to identify `undefined`, which I replaced by `checker.isUndefinedSymbol(symbol))` (maybe there is better solution), because the following code can be converted to an async function

```ts
function f() {
    return fetch('https://typescriptlang.org').then(undefined);
}
```

However, `Identifier` should refer to function declaration, the following code should not be converted to an async function

```ts
const ref = 1 // or undefined
function f1() {
    return fetch('https://typescriptlang.org').then(ref);
}
```

That's because the following code is not valid 

```ts
function f2() {
    return fetch('https://typescriptlang.org').then(1);
}
```

If identifier/prop access refers to a function, the function can be converted to an async function

```ts
const ref = function ()  {}
function f1() {
    return fetch('https://typescriptlang.org').then(ref);
}
```

or 

```ts
const ref = { x: function ()  {} }
function f1() {
    return fetch('https://typescriptlang.org').then(ref.x);
}
```

@sandersn Does that make sense?